packages/davinci-client/src/lib/client.store.ts

```diff
@@ -41,12 +41,12 @@ import type { RequestMiddleware } from './effects/request.effect.types.js';
  * @param {ConfigurationOptions} options - the configuration options for the client
  * @returns {Observable} - an observable client for DaVinci flows
  */
-export async function davinci({
+export async function davinci<ActionType extends ActionTypes = ActionTypes>({
   config,
   requestMiddleware,
 }: {
   config: DaVinciConfig;
-  requestMiddleware?: RequestMiddleware[];
+  requestMiddleware?: RequestMiddleware<ActionType>[];
```

Comment:
Just curious what this sort of typing means (`ActionType extends ActionTypes = ActionTypes`)?

Reply:
This is a generic, so `ActionType` is like a type variable. `extends` in this case means it kind of matches this type but maybe not fully.

So this allows for a little tighter type narrowing over the type. If I just pass in 'DavinciStart' instead of the entire ActionTypes union, typescript will know we are only operating on 'DavinciStart' and we arent actually using the rest of the types in the union of ActionType.

I called it ActionType for a more descriptive name, but commonly you may see `T` as the name of the generic

Reply:
what does the `= ActionTypes` part mean? I'm confused if this is interpreted more like `(ActionType extends ActionTypes) = ActionTypes` or `ActionType extends (ActionTypes = ActionTypes)`

Reply:
Sorry left that out, It's a default if a generic isn't passed. I kept it there for ease of use, and so that if you don't include the generic it has a safety to fallback to.